controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/Upgrader.java

```diff
@@ -78,16 +85,20 @@ private InstanceList instances(VersionStatus versionStatus) {
 
     private void cancelBrokenUpgrades(VersionStatus versionStatus) {
         // Cancel upgrades to broken targets (let other ongoing upgrades complete to avoid starvation)
-        InstanceList instances = instances(controller().readVersionStatus());
+        InstanceList instances = instances(deploymentStatuses(controller().readVersionStatus()));
         for (VespaVersion version : versionStatus.versions()) {
             if (version.confidence() == Confidence.broken)
                 cancelUpgradesOf(instances.upgradingTo(version.versionNumber()).not().with(UpgradePolicy.canary),
                                  version.versionNumber() + " is broken");
         }
     }
 
-    private void updateTargets(VersionStatus versionStatus, InstanceList instances, UpgradePolicy policy, OptionalInt targetMajorVersion) {
+    private void updateTargets(VersionStatus versionStatus, DeploymentStatusList deploymentStatuses, UpgradePolicy policy, OptionalInt targetMajorVersion) {
+        InstanceList instances = instances(deploymentStatuses);
         InstanceList remaining = instances.with(policy);
+        Instant failureThreshold = controller().clock().instant().minus(Duration.ofDays(5));
```

Comment:
Consider extracting the timeout as a constant.

Reply:
Yes ...